test/e2e/app-dir/dynamic-io/dynamic-io.routes.test.ts

```diff
@@ -206,6 +209,60 @@ describe('dynamic-io', () => {
     expect(message2).toEqual(json.message2)
   })
 
+  itSkipTurbopack(
+    'should prerender GET route handlers that have entirely cached io ("use cache")',
+    async () => {
+      let str = await next.render('/routes/use_cache-cached', {})
+      let json = JSON.parse(str)
+
+      let message1 = json.message1
+      let message2 = json.message2
+
+      if (isNextDev) {
+        expect(json.value).toEqual('at runtime')
+        expect(typeof message1).toBe('string')
+        expect(typeof message2).toBe('string')
+      } else {
+        expect(json.value).toEqual('at buildtime')
+        expect(typeof message1).toBe('string')
+        expect(typeof message2).toBe('string')
+      }
+
+      str = await next.render('/routes/use_cache-cached', {})
+      json = JSON.parse(str)
+
+      if (isNextDev) {
+        expect(json.value).toEqual('at runtime')
+        expect(message1).toEqual(json.message1)
+        expect(message2).toEqual(json.message2)
+      } else {
+        expect(json.value).toEqual('at buildtime')
+        expect(message1).toEqual(json.message1)
+        expect(message2).toEqual(json.message2)
+      }
+
+      // TODO: Edge is missing Server Manifest for routes.
```

Comment:
@unstubbable We can re-enable this once edge runtime has the correct server manifest initialized.